[aotinductor] Add AOTIModelRunner as a utility class

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110920
* __->__ #110891
* #110652

Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

desertfire committed:
[aotinductor] Add AOTIModelRunner as a utility class

Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add AOTIModelRunner as a utility class"


Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add AOTIModelRunner as a utility class"


Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add AOTIModelRunner as a utility class"


Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add AOTIModelRunner as a utility class"


Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

malfet commented on torch/csrc/inductor/aoti_model_runner_cuda.h:
```diff
@@ -0,0 +1,23 @@
+
+#include <torch/csrc/inductor/aoti_model_runner.h>
+
+namespace torch {
+namespace inductor {
```
Please, we live in C++17 world :)
```suggestion
namespace torch::inductor {
```

malfet commented on torch/csrc/inductor/aoti_model_runner_cuda.h:
```diff
@@ -0,0 +1,23 @@
+      AOTIProxyExecutorHandle proxy_executor_handle = nullptr);
+};
+
+} // namespace inductor
+} // namespace torch
```
Same as above
```suggestion
} // namespace torch::inductor
```

desertfire replied:
LOL, will change. While you are here, do you have any suggestion on how to fix this mobile build failure? [pull / linux-jammy-py3-clang12-mobile-build / build](https://hud.pytorch.org/pr/pytorch/pytorch/110891#17576311926) ([gh](https://github.com/pytorch/pytorch/actions/runs/6471105050/job/17576311926)) I want to exclude my newly added files from the mobile build, but I haven't figured out how to do that.

desertfire committed:
Update on "[aotinductor] Add AOTIModelRunner as a utility class"


Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Add AOTIModelRunner as a utility class"


Summary: Introduce a utility class AOTIModelRunner to take care of running an AOTInductor compiled model. It does things like dlopen a model, initialize the model container, setup inputs and outputs, and destroy the model container.

[ghstack-poisoned]

chenyang78 commented on torch/csrc/inductor/aoti_model_runner_cuda.h:
```diff
@@ -0,0 +1,21 @@
+
+  std::vector<at::Tensor> run(
+      std::vector<at::Tensor> inputs,
+      AOTInductorStreamHandle cuda_stream_handle = nullptr,
```
Instead of having the user to pass ```AOTInductorStreamHandle```, I am wondering if it would be better to just let her pass a ```cudaStream_t```. It might be slightly more user-friendly, because the user don't need to worry about what this ```AOTInductorStreamHandle``` is, nor does she need to convert a real ```cudaStream_t``` to a handle. Inside the ```run``` method, we can do the conversion for the user.

chenyang78 commented on torch/csrc/inductor/aoti_model_runner_cuda.h:
```diff
@@ -0,0 +1,21 @@
+  std::vector<at::Tensor> run(
+      std::vector<at::Tensor> inputs,
+      AOTInductorStreamHandle cuda_stream_handle = nullptr,
+      AOTIProxyExecutorHandle proxy_executor_handle = nullptr);
```
Same here - instead of passing a handle, could we pass a real ProxcyExecutor pointer?

chenyang78 commented on torch/csrc/inductor/aoti_model_runner_cuda.h:
```diff
@@ -0,0 +1,21 @@
+
+namespace torch::inductor {
+
+class TORCH_API AOTIModelRunnerCuda : public AOTIModelRunner {
```
AOTModelRunnerCuda inherits from AOTIModelRunner, which is guarded by two macros. It may be safer to add the same guard to aoti_model_runner_cuda.[h|cpp]?
